pkg/promql/engine.go

```diff
@@ -1109,23 +1109,28 @@ func (ev *evaluator) eval(expr parser.Expr) (parser.Value, storage.Warnings) {
 		var (
 			mat      Matrix
 			warnings storage.Warnings
+			err      error
 		)
 		parser.Inspect(expr, func(node parser.Node, path []parser.Node) error {
 			switch n := node.(type) {
 			/* Note that a MatrixSelector cascades down to it's VectorSelector in Inspect */
 			case *parser.VectorSelector:
-				warns, err := checkAndExpandSeriesSet(ev.ctx, n)
+				warnings, err = checkAndExpandSeriesSet(ev.ctx, n)
 				if err != nil {
-					return fmt.Errorf("topNode eval vector-selector: %w", err)
+					err = fmt.Errorf("topNode eval vector-selector: %w", err)
+					return err
```

Comment:
No we need this because this is a closure on the err defined in line 1112 and checked in line 1131. Unfortunately, parser.Inspect will not return the error